mapper/transaction.go

```diff
@@ -172,9 +171,50 @@ func CrossChainTransactions(
 			idx++
 		}
 	default:
-		panic("Unsupported transaction:" + reflect.TypeOf(t).String())
+		return nil, fmt.Errorf("Unsupported transaction: %s", reflect.TypeOf(t).String())
+	}
+	return ops, nil
+}
+
+func CrossChainTransactions(
+	avaxAssetID string,
+	block *ethtypes.Block,
+	AP5Activation uint64,
+) ([]*types.Transaction, error) {
+	transactions := []*types.Transaction{}
+
+	extra := block.ExtData()
+	if len(extra) == 0 {
+		return transactions, nil
+	}
+
+	// Initialize Atomic Transactions
+	atomicTxs := make([]*evm.Tx, 0)
+	if block.Time() < AP5Activation {
+		// Prior to Apricot Phase 5, there was only one atomic transaction per
+		// block.
+		tx := new(evm.Tx)
+		if _, err := codecManager.Unmarshal(extra, tx); err != nil {
+			return nil, err
+		}
+		atomicTxs = append(atomicTxs, tx)
+	} else {
+		if _, err := codecManager.Unmarshal(extra, &atomicTxs); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal atomic tx (AP5) due to %w", err)
+		}
+	}
+
+	ops := []*types.Operation{}
+	for _, tx := range atomicTxs {
+		txOps, err := crossChainTransaction(len(ops), avaxAssetID, tx)
+		if err != nil {
+			return nil, err
+		}
+		ops = append(ops, txOps...)
 	}
 
+	// TODO: migrate to using atomic transaction ID instead of marking as a block
```

Comment:
We may want to change this in this release, however, this would mean CB would need to re-index things.